prov/rxd/src/rxd_cq.c

```diff
@@ -380,27 +379,58 @@ static int rxd_match_tmsg(struct dlist_entry *item, const void *arg)
 			     attr->tag);
 }
 
+static void rxd_init_insert_unexp(struct rxd_ep *ep, struct dlist_entry *list,
+				  struct rxd_pkt_entry *pkt_entry,
+				  struct rxd_base_hdr *base_hdr,
+				  struct rxd_sar_hdr *sar_hdr,
+				  struct rxd_tag_hdr *tag_hdr,
+				  struct rxd_data_hdr *data_hdr,
+				  void *msg, size_t msg_size)
+{
+	struct rxd_unexp_msg *unexp_msg;
+
+	unexp_msg = calloc(1, sizeof(*unexp_msg));
+	unexp_msg->pkt_entry = pkt_entry;
+	unexp_msg->base_hdr = base_hdr;
+	unexp_msg->sar_hdr = sar_hdr;
+	unexp_msg->tag_hdr = tag_hdr;
+	unexp_msg->data_hdr = data_hdr;
+	unexp_msg->msg_size = msg_size;
+	unexp_msg->msg = msg;
+
+	dlist_init(&unexp_msg->pkt_list);
+	dlist_insert_tail(&unexp_msg->entry, list);
```

Comment:
I would remove the insertion here and just have the function init_unexp.  The insert makes the function weird, and functions don't like being weird.  Other functions make fun of them, they don't get to play in compiler games, etc.

Reply:
Well maybe we need to teach the other functions not to be such jerks. Stop victim shaming